src/blog/_posts/2024-02-15-cadence-non-deterministic-common-qa.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Cadence non-derministic errors common question Q&A (part 1)
+
+date: 2023-08-27
+author: Chris Qin
+authorlink: https://www.linkedin.com/in/chrisqin0610/
+---
+
+We will post a series of common Q&A blogs about Cadence non-deterministic errors to
+better help developers understand why they happens and how to troubleshoot at the first place.
+
+#### If I change code logic inside an Cadence activity (for example, my activity is calling database A but now I want it to call database B),  will it trigger an non-deterministic error?
+
+NO. This change will not trigger non-deterministic error.
+
+An Activity is the smallest unit of execution for Cadence and what happens inside activities are not recorded as historical events and therefore will not be replayed. In short, this change is deterministic and it is fine to modify logic inside activities.
+
+#### Does changing the activity parameter count make my workflow non-deterministic?
```

Comment:
Let's add the scenario about order of activities here since we are already talking about the activities. Also, adding a new activity without versioning is another common case. Rest can be included in Part 2

Reply:
Update the file to include more cases